Notepad-Sharp/Core/Logging/RuntimeLoggerService.cs

```diff
@@ -9,19 +7,48 @@ namespace NotepadSharp.Core.Logging
     /// </summary>
     public class RuntimeLoggerService : ILoggingService
     {
+        /// <summary>
+        /// Writes error message to log file.
+        /// </summary>
+        /// <param name="message"></param>
         public void Error(string message)
         {
-            MessageBox.Show(message, "Error");
+            Write("Error", message);
         }
 
+        /// <summary>
+        /// Writes informational message to log file.
+        /// </summary>
+        /// <param name="message"></param>
         public void Info(string message)
         {
-            MessageBox.Show(message, "Info");
+            Write("Info", message);
         }
 
+        /// <summary>
+        /// Writes warning message to log file.
+        /// </summary>
+        /// <param name="message"></param>
         public void Warn(string message)
         {
-            MessageBox.Show(message, "Warning");
+            Write("Warning", message);
+        }
+
+        private static void Write(string type, string message)
+        {
+            string _path = System.IO.Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+
+            try
+            {
+                using (System.IO.StreamWriter streamWriter = System.IO.File.AppendText(System.IO.Path.Combine(_path, "Log.txt")))
+                {
+                    streamWriter.WriteLine($"{System.DateTime.Now.ToLongTimeString()} {System.DateTime.Now.ToLongDateString()} [{type}]: {message}");
+                }
+            }
+            catch (System.Exception ex)
+            {
+                System.Windows.Forms.MessageBox.Show(ex.ToString(), "Error");
+            }
         }
```

Comment:
This method just tries to write messages along with the message type to the log file in the current working directory.

If an exception is encountered while trying to write to the log file, then the exception is shown to the user. This would be considered a fatal error scenario.